core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/ChangeStream.java

```diff
@@ -120,11 +135,11 @@ public boolean isOpen() {
    * Indicates whether or not any ChangeStreamListeners are currently running.
    * @return True if the ChangeStreamListeners are running
    */
-  public boolean listenersRunning() {
+  public boolean areListenersAttached() {
     if (runnerThread == null) {
       return false;
     }
-    return runnerThread.isAlive();
```

Comment:
[minor] this is super unlikely to happen, but this introduces the issue where if a listener is added (creating the runnerThread thread), but then removed (so the runner thread is still alive but listeners is empty), then nextEvent will not throw the IllegalStateException, even though it should because the runnerThread already called it.

I think this method should be called "isListenerThreadRunning", and it should just return `runnerThread.isAlive()`. It can be private or package private, since a user doesn't really need this method.